Sources/Swift/Tools/SentryClientReport.swift

```diff
@@ -0,0 +1,25 @@
+@objc
+@_spi(Private) public final class SentryClientReport: NSObject, SentrySerializable {
+    
+    @objc(initWithDiscardedEvents:dateProvider:)
+    public init(discardedEvents: [SentryDiscardedEvent], dateProvider: SentryCurrentDateProvider) {
```

Comment:
`l`: Maybe we could even remove the dependency to the `SentryCurrentDateProvider` here and only ask for the current date, which has the downside that everybody now needs to add `.date()` in the tests, and the HTTPTransport. I doesn't really matter cause both are OK.

Reply:
Yep, keep it as is then.